### PR TITLE
[Telemetry Operator] Make collection of k8s annotations and labels optional

### PR DESCRIPTION
**Pull request by dennis-ge**
**Description**

Changes proposed in this pull request:

- Add `keepAnnotations` and `dropLabels` CRD attributes in application input
- Those fields make the collection of kubernetes annotations and labels optional. Annotations are by default *not* collected and labels are by default collected.
- This is done using Fluent Bit [Nest](https://docs.fluentbit.io/manual/pipeline/filters/nest) and [record_modifier](https://docs.fluentbit.io/manual/pipeline/filters/record-modifier) filters.

**Related issue(s)**
Fixes https://github.com/kyma-project/kyma/issues/14788

### CODE REVIEW

**Review by lindnerby — APPROVED** [review]

File: tests/fast-integration/telemetry-test/telemetry-test.js [hunk 1]
```diff
@@ -179,4 +179,14 @@ describe('Telemetry Operator tests, prepare the environment', function() {
...
+
+  it('Should verify that Kubernetes annotations are not pushed', async () => {
+    const labels = '{namespace="mockserver", container="mockserver"}';
+    const responseBody = await queryLoki(labels, testStartTimestamp);
+    assert.isTrue(responseBody.data.result.length > 0, 'No mockserver logs present in Loki');
+    const entry = JSON.parse(responseBody.data.result[0].values[0][1]);
+
+    expect(entry.log).not.to.have.string('annotations');
+    expect(entry.log).to.have.string('labels');
+  });
```
Comment (skhalash) [inline tests/fast-integration/telemetry-test/telemetry-test.js:191]: Let's maybe create 2 acceptance tests: one for `KeepAnnotations` and one for `DropLabels`?

**Review by skhalash — CHANGES_REQUESTED** [review]

File: components/telemetry-operator/internal/fluentbit/config_builder.go [hunk 2]
```diff
@@ -33,11 +33,32 @@ name                  lua
 match                 %s.*
 script                /fluent-bit/scripts/filter-script.lua
 call                  kubernetes_map_keys`
+	LiftKubernetesKeysTemplate string = `
```
[the remark this answers is not in the document]
Comment (dennis-ge) [reply components/telemetry-operator/internal/fluentbit/config_builder.go:36]: yes, will fix it 👍

File: tests/fast-integration/telemetry-test/telemetry-test.js (hunk 1 above)
Comment (dennis-ge) [reply tests/fast-integration/telemetry-test/telemetry-test.js:191]: don't want to create an extra LogPipeline for this, will try to reuse the Loki LogPipeline

Comment (dennis-ge) [reply tests/fast-integration/telemetry-test/telemetry-test.js:191]: ⬆️ won't work as we remove all kubernetes keys in the Loki LogPipeline, see https://github.com/kyma-project/kyma/blob/main/resources/logging/charts/loki/templates/kyma-additions/logpipeline_loki.yaml#L16

File: tests/fast-integration/telemetry-test/telemetry-test.js [hunk 3]
```diff
@@ -169,17 +169,6 @@ describe('Telemetry Operator tests, prepare the environment', function() {
     assert.isTrue(logsPresent, 'No logs received by mockserver present in Loki');
   });
 
-  it('Include kyma-system namespace on loki pipeline ', async () => {
```
Comment (dennis-ge) [inline tests/fast-integration/telemetry-test/telemetry-test.js:172]: this test is not necessary anymore with https://github.com/kyma-project/kyma/pull/15030

File: tests/fast-integration/telemetry-test/telemetry-test.js [hunk 4]
```diff
@@ -189,4 +178,23 @@ describe('Telemetry Operator tests, prepare the environment', function() {
...
+
+  it('Should verify that Kubernetes labels are not pushed', async () => {
+    const httpPipeline = await getLogPipeline('http-mockserver');
+    httpPipeline.spec.input.application.keepAnnotations = true;
+    httpPipeline.spec.input.application.dropLabels = true;
+    await updateLogPipeline(httpPipeline);
+
+    await sleep(20 * 1000);
+    const updatedTestTimestamp = new Date().toISOString();
+    await sleep(20 * 1000);
```
Comment (dennis-ge) [inline tests/fast-integration/telemetry-test/telemetry-test.js:190]: needed there this long sleep so that logs actually show up

**Review by skhalash — DISMISSED** [review]

**Review by skhalash — APPROVED** [review]

**Pull request merged**